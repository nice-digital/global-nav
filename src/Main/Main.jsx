import React from "react";
import { BackToTop } from "./BackToTop/BackToTop";
import PropTypes from "prop-types";
import classnames from "classnames";

import styles from "./Main.module.scss";

export function Main(props) {
	const { children, className, ...rest } = props;

	return (
		<main className={classnames([styles.main, className])} {...rest}>
			{children}
			<BackToTop />
		</main>
	);
}

Main.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node,
	]).isRequired,
	className: PropTypes.string,
};

export default Main;
