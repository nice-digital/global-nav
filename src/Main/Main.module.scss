@import '~@nice-digital/nds-core/scss/core';

$back-to-top-height: rem(
  $nds-spacing-medium * 3
); // Line height + bottom and top padding

@supports (position: sticky) {
  .main {
    padding-bottom: $back-to-top-height; /* space for the back to top link to dock */
    position: relative;

    &.withPadding {
      padding-bottom: rem($nds-spacing-medium);
    }
  }
}
