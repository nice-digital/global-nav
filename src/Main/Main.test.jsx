import Main from "./Main";
import { render } from "@testing-library/react";

describe("Main", () => {
	it("Matches snapshot", () => {
		const { container } = render(
			<Main>
				<p>test</p>
			</Main>
		);
		expect(container).toMatchSnapshot();
	});
});
