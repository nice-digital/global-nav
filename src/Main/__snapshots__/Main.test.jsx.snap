// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Matches snapshot 1`] = `
<main
  className="main"
>
  <p>
    test
  </p>
  <BackToTop />
</main>
`;
